include/moveitem.h

```diff
@@ -297,3 +297,44 @@ int FreeSlotForItem(ItemClient* pItem)
 	}
 	return 0;
 }
+
+CItemLocation* FreeItemLocationForItem(CItemLocation* pFreeSlot, ItemClient* pItemClient)
+{
+	unsigned char  ucCurSize = 10;
+	ItemDefinition* pUnequipInfo = pItemClient->GetItemDefinition();
+	if (!pUnequipInfo) {
+		return nullptr;
+	}
```

Comment:
if pItemClient is non-null, then GetItemDefinition will always return non-null

Its not possible for an item to have no definition.